app/config.go

```diff
@@ -108,15 +109,27 @@ func subconfig(globalConfig *viper.Viper, subconfigKey string) *viper.Viper {
 }
 
 // DBConfig returns the db connection fixed config from the global config.
-// Panic in case of unmarshaling error
 func DBConfig(globalConfig *viper.Viper) (config *mysql.Config, err error) {
-	sub := subconfig(globalConfig, databaseConfigKey)
-	err = sub.Unmarshal(&config)
+	// Env variables are not loaded if the keys do not exist in the config file
+	// To fix this issue, instead of loading config files and overriding with env vars,
+	// we load all possible keys (with their default value), override with config files,
+	// and then environmenent variables.
+	emptyConfig := &map[string]interface{}{}
+	if err = mapstructure.Decode(mysql.NewConfig(), &emptyConfig); err != nil {
+		return // unexpected
```

Comment:
If it is really unexpected, it's better just to ignore the error (`_ = mapstructure.Decode(...)`)

Reply:
It is unexpected at runtime but if mysql.NewConfig() or mapstructure change their implem, it may fail. The "problem" if I ignore it is that the variable will keep its default value and may be undetected.